lib/interfaces/common.js

```diff
@@ -145,7 +145,16 @@ module.exports = function (suites, context, mocha) {
           throw createUnsupportedError('Pending test forbidden');
         }
         if (typeof opts.fn === 'function') {
-          opts.fn.call(suite);
+          const suiteResult = opts.fn.call(suite);
+          if (suiteResult instanceof Promise) {
+            errors.deprecate(
```

Comment:
On first glance I didn't like printing a **deprecation** message, since we never have supported async callbacks in `describe`. On the other hand I do like the deprecation message being cashed and printed only once. So we leave it as is.